drivers/platform/stm32/stm32_dma.c

```diff
@@ -245,12 +245,12 @@ int stm32_dma_xfer_abort(struct no_os_dma_desc* desc,
 
 	sdma_chan = chan->extra;
 
-	if (chan->irq_num)
-		ret = HAL_DMA_Abort_IT(sdma_chan->hdma);
-	else
-		ret = HAL_DMA_Abort(sdma_chan->hdma);
+	ret = HAL_DMA_Abort(sdma_chan->hdma);
```

Comment:
There are 2 changes here:
1. the abort type change
2. the hal error on aborting 2 times in a row

I think you should split them in 2 commits.
1st commit, I'm not sure I understand the problem.
2nd commit, ok, seems like a legit fix.